src/qt/spinbutt.cpp

```diff
@@ -34,12 +35,32 @@ void wxQtSpinButton::valueChanged(int value)
     wxSpinButton *handler = GetHandler();
     if ( handler )
     {
-        wxCommandEvent event( wxEVT_SPIN, handler->GetId() );
+        wxSpinEvent event( wxEVT_SPIN, handler->GetId() );
         event.SetInt( value );
         EmitEvent( event );
     }
 }
 
+void wxQtSpinButton::stepBy(int steps)
+{
+    wxSpinButton* handler = GetHandler();
+    if ( !handler )
+        return;
+
+    int eventType = (steps < 0) ? wxEVT_SPIN_DOWN : wxEVT_SPIN_UP;
+    wxSpinEvent directionEvent(eventType, handler->GetId());
+    directionEvent.SetPosition(value());
+    directionEvent.SetInt(value() + steps * singleStep());
+    directionEvent.SetEventObject(handler);
+
+    handler->HandleWindowEvent(directionEvent); // should return value be checked here?
```

Comment:
No, this event can't be vetoed, so:
```suggestion
    handler->HandleWindowEvent(directionEvent);
```

Reply:
I've removed the comment, but the event _can_ be vetoed, see the docs and the very next line of code that actually checks for it being allowed.

Reply:
Oops, I'm awfully sorry, I totally forgot that it could be vetoed but it indeed can and it does work in at least wxMSW and wxGTK (I'll add a small test for this to the widgets sample).

And so, to completely change what I wrote, I think the return value should indeed be checked as if a handler calls `Veto()` and then `Skip()`, its veto presumably shouldn't count (and, again, this is indeed how it works currently).

So the code should do something like

    if ( !handler->HandleWindowEvent(dirEvent) || dirEvent.IsAllowed() )
         QSpinBox::stepBy(...);

Could you please update it to work like this and check that it works as expected with my changes to the widgets sample (sorry but I just don't have time to write a proper unit test for this right now)?

TIA and sorry again for the blatant disinformation above.

Reply:
@vadz Updated with your suggestion. It did work as expected with it.